src/definitions/namespaces/org_iso_18013_5_1/driving_privileges.rs

```diff
@@ -16,10 +20,61 @@ impl From<DrivingPrivileges> for Cbor {
     }
 }
 
+/// The specifications of vehicle categories and restrictions
+#[derive(Clone, Eq, PartialEq, EnumString, Debug, EnumVariantNames, AsRefStr)]
+pub enum VehicleCategoryCode {
+    /// Motorcycles
+    A,
+    /// Light vehicles
+    B,
+    /// Goods vehicles
+    C,
+    // Passenger vehicles
+    D,
+    /// Light vehicles with trailers
+    BE,
+    /// Goods vehicles with trailers
+    CE,
+    /// Passenger vehicles with trailers
+    DE,
+    ///Mopeds
+    AM,
+    /// Light motorcycles
+    A1,
+    /// Medium motorcycles
+    A2,
+    /// Light vehicles
+    B1,
+    /// Medium sized goods vehicles
+    C1,
+    /// Medium sized passenger vehicles (e.g.minibuses)
+    D1,
+    /// Medium sized goods vehicles with trailers
+    C1E,
+    /// Medium sized passenger vehicles (e.g. minibuses) with trailers
+    D1E,
+}
+
+impl From<VehicleCategoryCode> for Cbor {
+    fn from(c: VehicleCategoryCode) -> Cbor {
+        Cbor::Text(c.as_ref().to_string())
+    }
+}
+
+impl crate::definitions::traits::FromJson for VehicleCategoryCode {
+    fn from_json(v: &serde_json::Value) -> Result<Self, FromJsonError> {
+        v.as_str()
+            .map(str::to_uppercase)
+            .map(|s| s.parse::<VehicleCategoryCode>())
+            .unwrap_or(Err(strum::ParseError::VariantNotFound))
+            .map_err(|_| FromJsonError::Missing)
```

Comment:
Please use more accurate error variants, and leverage the existing `FromJson` implementations for this:
```suggestion
        String::from_json(v)?
            .parse()
            .map_err(FromJson::Parsing)
```